core/network/src/service.rs

```diff
@@ -236,27 +239,27 @@ impl<B: BlockT + 'static, S: NetworkSpecialization<B>> Service<B, S> {
 	pub fn on_block_imported(&self, hash: B::Hash, header: B::Header) {
 		let _ = self
 			.protocol_sender
-			.send(ProtocolMsg::BlockImported(hash, header));
+			.unbounded_send(ProtocolMsg::BlockImported(hash, header));
```

Comment:
Yes and no. The message is only pushed on the channel if we import a block. If we're accepting all block imports, we're probably going to run out of memory because of the blocks and not because of this small message.
The less appealing answer is that we're clearly not protected against DoS at the moment.